packages/transaction-controller/src/TransactionController.ts

```diff
@@ -890,6 +923,7 @@ export class TransactionController extends BaseController<
   ): Promise<string> {
     const transactionId = transactionMeta.id;
     let resultCallbacks: AcceptResultCallbacks | undefined;
+    const { actionId } = transactionMeta;
 
     try {
       const acceptResult = await this.requestApproval(transactionMeta, {
```

Comment:
One key addition I think we need is to skip additional approvals if we are a duplicate call.

The extension does this by passing an `isExisting` argument to this method, skipping the approval if true, and then waiting for a suitable event before resolving the promise.

Unless of course the status of the meta is already final in which case we resolve immediately.

Reply:
Included `isExisting` argument in `processApproval` and skipping the approval creation if `true` as in the extension.

Reply:
Great! Now it's waiting for `${transactionMeta.id}:finished` event as in the extension. It also checks if the transaction is not finished to skip sign and submit and instead hold the promised status is finished.